### PR TITLE
Use `postcss-selector-parser` to handle all cases with the `&` selector

### PR DESCRIPTION
**Pull request by ExE-Boss**
This makes it possible to use selectors like:
```css
.wrapper > & {
    /* properties when the scope is wrapped in `.wrapper` */
}

& > .stuff {
    /* properties when `.stuff` is a direct descendant of the scope */
}

&.custom {
    /* properties when the scope also has the class `.custom` */
}
```
which get converted to:
```css
.wrapper > .scope {
    /* properties when the scope is wrapped in `.wrapper` */
}

.scope > .stuff {
    /* properties when `.stuff` is a direct descendant of the scope */
}

.scope.custom {
    /* properties when the scope also has the class `.custom` */
}
```

---

review?(@pazams)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,9 +5,9 @@ before_install:
   - "sh -e /etc/init.d/xvfb start"
 
 node_js:
-  - '0.10'
+  - '4'
 
 before_script:
-  - 'npm install'
+  - 'npm run lint'
```
Comment (ExE-Boss) [inline .travis.yml]: `npm install` is already run before this stage, so the build used to call `npm install` twice.

File: .eslintrc [hunk 2]
```diff
@@ -1,15 +1,15 @@
 {
+    "extends": "eslint:recommended",
     "rules": {
         "no-unused-expressions": [0],
         "no-underscore-dangle":  [0],
-        "no-reserved-keys":      [2],
```
Comment (ExE-Boss) [inline .eslintrc]: `eslint:recommended` already contains this.

**Comment by pazams** [comment]
@ExE-Boss sorry I missed this PR 🙏 
I just did a quick pass, and overall this looks great 👏 🙌 
I'll do a proper review in the next 3-4 days

**Comment by ExE-Boss** [comment]
Looks like the build broke as a result of Travis moving to **Ubuntu 16 Xenial**.

I’m providing a fix in #14

**Review by pazams — APPROVED** [review]
Amazing work 🙇 
Apologies for delayed review and merge :)

**Pull request merged**